### PR TITLE
changes to avoid deprecated API's

### PR DESCRIPTION
**Pull request by wjwwood**
Connects to ros2/rclcpp#713

### CODE REVIEW

File: test_rclcpp/test/test_executor.cpp [hunk 1]
```diff
@@ -205,15 +205,16 @@ TEST(CLASSNAME(test_executor, RMW_IMPLEMENTATION), notify) {
         executor.cancel();
       };
 
+    auto qos = rclcpp::QoS(rclcpp::QoSInitialization::from_rmw(rmw_qos_profile_default));
     auto subscription = node->create_subscription<test_rclcpp::msg::UInt32>(
       "test_executor_notify_subscription",
-      sub_callback,
-      rmw_qos_profile_default);
+      qos,
```
Comment (ivanpauno) [inline test_rclcpp/test/test_executor.cpp]: Replace this line with `10,` and delete L208.

File: test_security/test/test_secure_publisher.cpp [hunk 2]
```diff
@@ -30,11 +30,8 @@ int8_t attempt_publish(
 {
   auto start = std::chrono::steady_clock::now();
 
-  rmw_qos_profile_t custom_qos_profile = rmw_qos_profile_default;
-  custom_qos_profile.depth = messages.size();
-
   auto publisher = node->create_publisher<T>(
-    topic_name, custom_qos_profile);
+    topic_name, rclcpp::QoS(rclcpp::KeepLast(messages.size())));
```
Comment (ivanpauno) [inline test_security/test/test_secure_publisher.cpp]: Maybe, replace: `rclcpp::QoS(rclcpp::KeepLast(messages.size()))`
with: `messages.size()`

File: test_security/test/test_secure_subscriber.cpp [hunk 3]
```diff
@@ -65,11 +65,9 @@ rclcpp::SubscriptionBase::SharedPtr attempt_subscribe(
       rclcpp::shutdown();
     };
 
-  rmw_qos_profile_t custom_qos_profile = rmw_qos_profile_default;
-  custom_qos_profile.depth = expected_messages.size();
+  auto qos = rclcpp::QoS(rclcpp::KeepLast(expected_messages.size()));
 
-  auto subscriber = node->create_subscription<T>(
-    topic_name, callback, custom_qos_profile);
+  auto subscriber = node->create_subscription<T>(topic_name, qos, callback);
```
Comment (ivanpauno) [inline test_security/test/test_secure_subscriber.cpp]: Replace `qos` with `expected_messages.size()` and delete L68.

File: test_communication/test/test_publisher.cpp [hunk 4]
```diff
@@ -29,11 +29,9 @@ void publish(
 {
   auto start = std::chrono::steady_clock::now();
 
-  rmw_qos_profile_t custom_qos_profile = rmw_qos_profile_default;
-  custom_qos_profile.depth = messages.size();
+  auto qos = rclcpp::QoS(rclcpp::KeepLast(messages.size()));
 
-  auto publisher = node->create_publisher<T>(
-    std::string("test/message/") + message_type, custom_qos_profile);
+  auto publisher = node->create_publisher<T>(std::string("test/message/") + message_type, qos);
```
[the remark this answers is not in the document]
Comment (wjwwood) [reply test_communication/test/test_publisher.cpp]: I'm trying to do in place changes as much as possible, again, I'm going to pass on stylistic changes to get this finished.

**Pull request merged**